### PR TITLE
Fixed reading/writing of contig names in VCF header

### PR DESCRIPTION
**Pull request by esiragusa**
Note that parsing could be smarter: now it expects ID as first field. Contig lengths could also be parsed.

### CODE REVIEW

**Comment by esiragusa** [comment]
Okay a couple of VCF test files must be updated as the header format changed.

Mason and Gustaf seem to have this functionality implemented on their own (!) as they now output twice contig header lines. @holtgrewe @ktrappe suggestions?

**Comment by ktrappe** [comment]
I think that was the way you had to do it with the (old) I/O module. Is this only an internal seqan update or have the VCF specs changed somehow (I don't see a difference at first glance when I open the VCF spec 4.0).

File: include/seqan/vcf_io/write_vcf.h [hunk 1]
```diff
@@ -61,6 +61,15 @@ writeHeader(TTarget & target,
         writeValue(target, '\n');
     }
 
+    for (unsigned i = 0; i < length(contigNames(context)); ++i)
```
Comment (rrahn) [inline include/seqan/vcf_io/write_vcf.h]: Why not using range based for loop here?

**Comment by rrahn** [comment]
No, it just seems, regarding the submitted changes, that we never wrote out the contig names. 
Probably you have written your own method to do this in Gustaf/Mason?

File: include/seqan/vcf_io/write_vcf.h (hunk 1 above)
Comment (rrahn) [inline include/seqan/vcf_io/write_vcf.h]: Really 😞

Comment (rrahn) [inline include/seqan/vcf_io/write_vcf.h]: Than just leave it.

**Comment by ktrappe** [comment]
Yes, I have written my own method to write VCF headers and records:
https://github.com/seqan/seqan/blob/develop/apps/gustaf/msplazer_out.h#L971
and I do fill in the sequence names there too
https://github.com/seqan/seqan/blob/develop/apps/gustaf/msplazer_out.h#L1009

**Comment by esiragusa** [comment]
Closing in favor of #1941.

**Pull request closed without merging**